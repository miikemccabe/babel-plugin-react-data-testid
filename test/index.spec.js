import pluginTester from "babel-plugin-tester";

import reactDataTestId from "../src";

pluginTester({
  plugin: reactDataTestId,
  pluginName: "react-data-testid",
  filename: __filename,
  babelOptions: {
    babelrc: true,
    filename: __filename,
  },
  tests: [
    {
      title: `Doesn't change if there is nothing exported`,
      code: `<div>Hello</div>;`,
      snapshot: true
    },
    {
      title: `Doesn't change if the parent is a fragment`,
      code: `export default test = <><div>Hello</div></>;`,
      snapshot: true
    },
    {
      title: `Adds a data-testid with the value of the default export variable name`,
      code: `export default test = <div>Hello</div>;`,
      snapshot: true
    },
    {
      title: `Concatonates the data-testid if one already exists`,
      code: `export default test = <div data-testid="original">Hello</div>;`,
      snapshot: true
    },
    {
      title: `The test id can be changed via config`,
      pluginOptions: {
        attributeName: "data-test",
      },
      code: `export default test = <div>Hello</div>;`,
      snapshot: true
    },
    {
      title: `Adds a data-testid for arrow functions with explicit return`,
      fixture: "__fixtures__/arrow-function/implicit-return.js",
      snapshot: true
    },
    {
      title: `Adds a data-testid for arrow functions with explicit return`,
      fixture: "__fixtures__/arrow-function/explicit-return.js",
      snapshot: true
    },
    {
      title: `Doesn't add attribute to a nested map`,
      fixture: "__fixtures__/arrow-function/with-nested-map.js",
      snapshot: true
    },
  ],
});
