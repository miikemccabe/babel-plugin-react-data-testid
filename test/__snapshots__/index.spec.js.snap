// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-data-testid Adds a data-testid for arrow functions with explicit return: Adds a data-testid for arrow functions with explicit return 1`] = `

const MyComponent = () => <div>Hello</div>;

      ↓ ↓ ↓ ↓ ↓ ↓

const MyComponent = () => <div>Hello</div>;


`;

exports[`react-data-testid Adds a data-testid for arrow functions with explicit return: Adds a data-testid for arrow functions with explicit return 2`] = `

const MyComponent = () => {
  return <div>Hello</div>;
}

      ↓ ↓ ↓ ↓ ↓ ↓

const MyComponent = () => {
  return <div>Hello</div>;
};


`;

exports[`react-data-testid Adds a data-testid with the value of the default export variable name: Adds a data-testid with the value of the default export variable name 1`] = `

export default test = <div>Hello</div>;

      ↓ ↓ ↓ ↓ ↓ ↓

Object.defineProperty(exports, "__esModule", {
  value: true,
});
exports.default = void 0;

var _default = (test = <div>Hello</div>);

exports.default = _default;


`;

exports[`react-data-testid Concatonates the data-testid if one already exists: Concatonates the data-testid if one already exists 1`] = `

export default test = <div data-testid="original">Hello</div>;

      ↓ ↓ ↓ ↓ ↓ ↓

Object.defineProperty(exports, "__esModule", {
  value: true,
});
exports.default = void 0;

var _default = (test = <div data-testid="original">Hello</div>);

exports.default = _default;


`;

exports[`react-data-testid Doesn't add attribute to a nested map: Doesn't add attribute to a nested map 1`] = `

export default Sizes = ({ sizes, selected, onClickSize }) => (
  <ul className={styles.sizes}>
    {sizes.map(({ inStock, size, sku, sizeLabel }, index) => (
      <li key={size} value={size} data-sku={sku}>
        {inStock ? (
          <button
            type="button"
            onClick={event => {
              event.preventDefault();
              onClickSize(size);
            }}
            className={size === selected ? styles.selectedSize : ''}
            aria-label={\`size \${sizeLabel}\`}
          >
            {sizeLabel}
          </button>
        ) : (
          <div className={styles.disabled}>{sizeLabel}</div>
        )}
      </li>
    ))}
  </ul>
);

      ↓ ↓ ↓ ↓ ↓ ↓

Object.defineProperty(exports, "__esModule", {
  value: true,
});
exports.default = void 0;

var _default = (Sizes = ({ sizes, selected, onClickSize }) => (
  <ul data-testid="sizes" className={styles.sizes}>
    {sizes.map(({ inStock, size, sku, sizeLabel }, index) => (
      <li key={size} value={size} data-sku={sku}>
        {inStock ? (
          <button
            type="button"
            onClick={(event) => {
              event.preventDefault();
              onClickSize(size);
            }}
            className={size === selected ? styles.selectedSize : ""}
            aria-label={\`size \${sizeLabel}\`}
          >
            {sizeLabel}
          </button>
        ) : (
          <div className={styles.disabled}>{sizeLabel}</div>
        )}
      </li>
    ))}
  </ul>
));

exports.default = _default;


`;

exports[`react-data-testid Doesn't change if the parent is a fragment: Doesn't change if the parent is a fragment 1`] = `

export default test = <><div>Hello</div></>;

      ↓ ↓ ↓ ↓ ↓ ↓

Object.defineProperty(exports, "__esModule", {
  value: true,
});
exports.default = void 0;

var _default = (test = (
  <>
    <div>Hello</div>
  </>
));

exports.default = _default;


`;

exports[`react-data-testid Doesn't change if there is nothing exported: Doesn't change if there is nothing exported 1`] = `

<div>Hello</div>;

      ↓ ↓ ↓ ↓ ↓ ↓

<div>Hello</div>;


`;

exports[`react-data-testid The test id can be changed via config: The test id can be changed via config 1`] = `

export default test = <div>Hello</div>;

      ↓ ↓ ↓ ↓ ↓ ↓

Object.defineProperty(exports, "__esModule", {
  value: true,
});
exports.default = void 0;

var _default = (test = <div>Hello</div>);

exports.default = _default;


`;
